src/shaders/_prelude_shadow.fragment.glsl

```diff
@@ -0,0 +1,119 @@
+#ifdef RENDER_SHADOWS
+
+uniform sampler2D u_shadowmap_0;
+uniform sampler2D u_shadowmap_1;
+uniform float u_shadow_intensity;
+uniform float u_texel_size;
+uniform vec2 u_cascade_distances;
+
+float shadow_sample_1(vec2 uv, float compare) {
+    return step(unpack_depth(texture2D(u_shadowmap_1, uv)), compare);
+}
+
+float shadow_sample_0(vec2 uv, float compare) {
+    return step(unpack_depth(texture2D(u_shadowmap_0, uv)), compare);
+}
+
+float shadow_occlusion_1(vec4 pos, float bias) {
+    pos.xyz /= pos.w;
+    pos.xy = pos.xy * 0.5 + 0.5;
+    float fragDepth = min(pos.z, 0.999) - bias;
+    return shadow_sample_1(pos.xy, fragDepth);
+}
+
+float shadow_occlusion_0(vec4 pos, float bias) {
+    pos.xyz /= pos.w;
+    pos.xy = pos.xy * 0.5 + 0.5;
+    float fragDepth = min(pos.z, 0.999) - bias;
```

Comment:
why is `min(pos.z, 0.999)` required?

Reply:
This is probably not needed. I will try removing it.